### PR TITLE
Create workflows

### PR DESCRIPTION
**Pull request by mavaylon1**

### CODE REVIEW

**Comment by mavaylon1** [comment]
Fix #18, #10 , #11

File: docs/source/overview.rst [hunk 1]
```diff
@@ -42,4 +42,4 @@ TODO
...
 - Handling of  external links on export is not yet fully implemented and is missing a few corner cases
-- Here we update the PyNWB test harness to add ZarrIO to the rountrip tests, which in turn runs all HDF5 roundtrip tests also for Zarr. This requires changing the test harness in PyNWB, instead it would be useful to be able to "inject" new I/O backends in the test harness so that we can specify those tests here, rather than implementing this in PyNWB and making PyNWB dependent on hdmf-zarr (see https://github.com/NeurodataWithoutBorders/pynwb/pull/1018/files#diff-ebaf216d6ea84b72ee9ac17ab17b6e5d5ed8a5f3bbd09ee11d9ee6969c2fc294 )
\ No newline at end of file
+- Here we update the PyNWB test harness to add ZarrIO to the rountrip tests, which in turn runs all HDF5 roundtrip tests also for Zarr. This requires changing the test harness in PyNWB, instead it would be useful to be able to "inject" new I/O backends in the test harness so that we can specify those tests here, rather than implementing this in PyNWB and making PyNWB dependent on hdmf-zarr (see https://github.com/NeurodataWithoutBorders/pynwb/pull/1018/files)
```
Comment (oruebel) [inline docs/source/overview.rst]: The main reason I had used the other link is because the main changes this TODO item is referring to are the changes in ``tests/integration/ui_write/base.py`` as part of that PR (as well as the changes in ``tests/integration/hdf5/test_modular_storage.py``). The other changes in that PR (i.e., the additions for ``docs/notebooks/zarr_file_conversion_test.ipynb`` and ``src/pynwb/__init__.py`` are already part of the hdmf-zarr library. However, I can see that using the more specific link is confusing and the intent is not clear. Maybe we should just update this note to clarify what the relevant files from this PR are..

File: docs/source/overview.rst [hunk 2]
```diff
@@ -41,4 +41,5 @@ TODO
...
\ No newline at end of file
+=======
+- Here we update the PyNWB test harness to add ZarrIO to the rountrip tests, which in turn runs all HDF5 roundtrip tests also for Zarr. This requires changing the test harness in PyNWB, instead it would be useful to be able to "inject" new I/O backends in the test harness so that we can specify those tests here, rather than implementing this in PyNWB and making PyNWB dependent on hdmf-zarr. See the files ``tests/integration/ui_write/base.py`` and ``tests/integration/hdf5/test_modular_storage.py`` as part of `PyNWB #1018 <https://github.com/NeurodataWithoutBorders/pynwb/pull/1018/files>`_ for details (the other parts of the this PR have already been ported to *hdmf-zarr*).
```
Comment (oruebel) [inline docs/source/overview.rst]: The ``=======`` in Line 44 looks like it may be a leftover from a merge conflict and should be removed

File: test.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import re
+import argparse
+import logging
+import os.path
+import os
+import sys
+import traceback
+import unittest
+
+flags = {'hdmf': 1, 'example': 4}
```
Comment (oruebel) [inline test.py]: Instead of ``'hdmf'`` I think we should rename to the flag to ``'unit'``

**Comment by oruebel** [comment]
This looks good to me. @rly could you review this PR since you are more familiar with the CI.

File: .github/workflows/black.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+name: black
+
+on: pull_request
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (rly) [inline .github/workflows/black.yml:9]: These actions have newer versions. Can you update these?

**Comment by rly** [comment]
I recommend keeping `requirements-min.txt` from HDMF and testing with it. They should mirror the minimum requirements set in `setup.py`. We should test those in the CI (please revert some of the changes where you removed py3.7-minimum)

**Comment by mavaylon1** [comment]
@oruebel So the the gallery tests for 3.7 minimum are using an older version of dandi. I've tried to set a dandi minimum version but that breaks other parts. 
ex)
I added dandi>= 0.46.1 to the requirements-doc.txt which led to 
1. Run code coverage / macos-latest failing when it was passng prior.
2. The same gallery 3.7 minimum tests failing but for a new reason. ----> "FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt' ". 

For more details: checkout commit df51116

Any thoughts?

This is for the doc "plot_convert_nwb.py" line 19 showing an error in the init that seems to be fixed in newer dandi versions.

**Review by oruebel — APPROVED** [review]
Looks good to me
